### PR TITLE
More detailed logic for when overwriting is OK

### PR DESCRIPTION
**Pull request by drevell**
The logic in this PR is to use `overwrite=true` when writing to the scratch directory. This allows multiple include actions to touch the same set of files, if the template author wishes.

This PR also improves error messages in some edge cases.

This PR also forbids overwriting files with directories and vice versa, because that would be confusing for both template developers and abc code maintainers.

In an upcoming PR, we will set `overwrite=$(value_of_--force_overwrite_flag)` when writing to the final output directory.

### CODE REVIEW

**Review by yolocs — COMMENTED** [review]
From the description, it sounds like we'd default to not allow overwrite? I'm not clear when we want to preserve (the content in) the scratch folder.

**Comment by drevell** [comment]
> From the description, it sounds like we'd default to not allow overwrite? I'm not clear when we want to preserve (the content in) the scratch folder.

We never want to preserve the content of the scratch folder. Therefore, we use overwrite=true in actionInclude(). In the future, we'll want to use overwrite=$(value_of_overwrite_flag) when writing to the final destination directory (not the scratch directory). I'll update the PR description to try to make this more clear.

File: templates/commands/render.go [hunk 1]
```diff
@@ -391,6 +391,15 @@ func copyRecursive(pos *model.ConfigPos, from, to string, rfs renderFS) error {
 		if !d.IsDir() {
 			dirToCreate = filepath.Dir(dst)
 		}
+
+		dirInfo, err := rfs.Stat(dirToCreate)
+		if err != nil {
+			if !os.IsNotExist(err) {
+				return model.ErrWithPos(pos, "Stat(): %w", err) //nolint:wrapcheck
+			}
+		} else if !dirInfo.Mode().IsDir() {
```
Comment (sethvargo) [inline templates/commands/render.go:400]: Can `.Mode`() ever return nil?

**Review by sethvargo — APPROVED** [review]

File: templates/commands/render_action_include.go [hunk 2]
```diff
@@ -46,7 +46,12 @@ func actionInclude(ctx context.Context, i *model.Include, sp *stepParams) error
 			return fmt.Errorf("Stat(): %w", err)
 		}
 
-		if err := copyRecursive(p.Pos, absSrc, absDst, sp.fs); err != nil {
+		// Allow later includes to replace earlier includes in the scratch
+		// directory. This doesn't affect whether files in the final destination
+		// directory will be overwritten; that comes later.
+		const overwrite = true
```
Comment (yolocs) [inline templates/commands/render_action_include.go:52]: Thank you!

**Review by yolocs — APPROVED** [review]

**Pull request merged**